server/app/middlewares/websocket_backend.rb

```diff
@@ -135,16 +137,18 @@ def handle_rpc_response(data)
   def handle_rpc_request(ws, data)
     client = client_for_ws(ws)
     if client
-      @rpc_server.async.handle_request(ws, client[:grid_id].to_s, data)
+      @queue << [ws, client[:grid_id].to_s, data]
     end
   end
 
   # @param [Faye::WebSocket::Event] ws
   # @param [Array] data
   def handle_rpc_notification(ws, data)
+    return if @queue.size > 800 # too busy to handle notifications
```

Comment:
Needs logging of drops... we really have to make these kind of overload situations easier to diagnose as well.

```ruby
if @queue.size > 800
  @dropped += 1
  return
end
```

And then periodically warn on `@dropped > 0` and reset it?